app/domain/authentication/authn_jwt/functions.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Authentication
+  module AuthnJwt
+    # Joins "prefix" and "main" as a path, with exactly one slash (/).  If
+    # prefix ends with a slash or host begins with one, they're removed to avoid
+    # duplicate slashes.
+    class << self
+      def path_join(*parts)
+        parts.reduce { |m, x| path_join_two(m, x) }
```

Comment:
Maybe m,x can be replaced with text and concated? or other meaningful variable names?